include/nbl/asset/ICPUDescriptorSet.h

```diff
@@ -241,34 +262,58 @@ class NBL_API ICPUDescriptorSet final : public IDescriptorSet<ICPUDescriptorSetL
 			--_levelsBelow;
 			if (m_layout->isAnyDependencyDummy(_levelsBelow))
 				return true;
-			for (auto it = m_descriptors->begin(); it != m_descriptors->end(); it++)
+
+			for (uint32_t t = 0u; t < EDT_COUNT; ++t)
 			{
-				auto descriptor = it->desc.get();
-				if (!descriptor)
+				const auto type = static_cast<E_DESCRIPTOR_TYPE>(t);
+				const auto descriptorCount = m_layout->getTotalDescriptorCount(type);
+				if (descriptorCount == 0ull)
 					continue;
 
-				switch (descriptor->getTypeCategory())
+				auto descriptors = m_descriptors[type]->begin();
+				assert(descriptors);
+
+				for (uint32_t i = 0u; i < descriptorCount; ++i)
 				{
-				case IDescriptor::EC_BUFFER:
-					if (static_cast<ICPUBuffer*>(descriptor)->isAnyDependencyDummy(_levelsBelow))
-						return true;
-					break;
-				case IDescriptor::EC_IMAGE:
-					if (static_cast<ICPUImageView*>(descriptor)->isAnyDependencyDummy(_levelsBelow))
-						return true;
-					if (it->image.sampler && it->image.sampler->isAnyDependencyDummy(_levelsBelow))
-						return true;
-					break;
-				case IDescriptor::EC_BUFFER_VIEW:
-					if (static_cast<ICPUBufferView*>(descriptor)->isAnyDependencyDummy(_levelsBelow))
-						return true;
-					break;
+					switch (descriptors[i]->getTypeCategory())
```

Comment:
you already know the type category, shouldn't need to query it (derive, since you know `t`)

Reply:
Resolved.